edgecontext/req_context.go

```diff
@@ -47,6 +52,23 @@ func (e *EdgeRequestContext) AttachToContext(ctx context.Context) context.Contex
 	return ctx
 }
 
+// AttachHTTPHeader attaches the header to the http Headers
+func (e *EdgeRequestContext) AttachHTTPHeader(h http.Header) {
+	h.Add(httpbp.EdgeContextHeader, e.header)
+}
+
+// AttachSignedHTTPHeader attaches the header to the http Headers along with
+// a signature header.
+func (e *EdgeRequestContext) AttachSignedHTTPHeader(h http.Header, signer Signer) error {
```

Comment:
I think instead of asking the users to implement `Signer`, we should actually take care of that? We already provide [`httpbp.TrustHeaderSignature`](https://pkg.go.dev/github.com/reddit/baseplate.go/httpbp?tab=doc#TrustHeaderSignature), and this really should just be the reverse of that (or at least, we should provide an implementation of `Signer` in `httpbp` package).